### DISCUSSION TITLE
Add email notification support for failed processing

### DISCUSSION THREAD

**Discussion by engram-design** [engram-design/FeedMe#877] [Ideas]

**Comment by terryupton** [comment]
+1

**Comment by nathanielperales** [comment]
+1

**Comment by Coysh** [comment]
+1
Or even a way to continue over a failed import (i.e. I import thousands of users from a CRM - if 1  email is invalid, the whole import will fail)

**Comment by ursbraem** [comment]
+1

**Comment by anchovy** [comment]
Another solution would be to add an event for when a log type of _error_ is written.
This way developers can listen for this event and process the information as required - email, post to external error service etc.

Would love to see this implemented in some sort of way.
Gives developers the flexibility to manage the error in any way required.

For example....

```
use craft\feedme\events\FeedEvent;
use craft\feedme\services\Feeds;
use yii\base\Event;

Event::on(Feeds::class, Feeds::EVENT_PROCESSING_ERROR, function(FeedEvent $event) {

    // When this occurred
    $date = $event->date;

    // The type of log, in case this isn't just limited to error types.
    $type = $event->type;

    // Get the 'message'
    $message = $event->message;
    
    // Do something with this message
    ...

});
```
